samples: nrf9160: add sample for nRF Cloud FOTA via REST (#6035)
nrfconnect/sdk-nrf: samples/nrf9160/nrf_cloud_rest_fota/README.rst

This is useful when initially provisioning and associating a device on nRF Cloud.
You only need to do this once for each device.

When you start up your device and have not requested JITP, the sample will ask if FOTA should be enabled in the device's shadow.
Press the button configured by :kconfig:`CONFIG_REST_FOTA_BUTTON_EVT_NUM` to perform the shadow update.
Depending on how you provisioned your device with nRF Cloud, this step might not be needed.
You only need to do this once for each device.

If you have enabled the :kconfig:`CONFIG_REST_FOTA_ENABLE_LED` option, an LED configured by the :kconfig:`CONFIG_REST_FOTA_LED_NUM` option indicates the state of the connection to the LTE network.

Configuration
*************
|config|

Configuration options
=====================

Check and configure the following configuration options for the sample:

.. kconfig:: CONFIG_REST_FOTA_JOB_CHECK_RATE_MIN

CONFIG_REST_FOTA_JOB_CHECK_RATE_MIN - Update check rate
   This configuration option defines how often the sample checks for FOTA updates.

> 

Reviewer: Remove extra line.

.. _nrf_cloud_rest_fota:

nRF9160: nRF Cloud REST FOTA
#############################

.. contents::
   :local:
   :depth: 2

The REST FOTA sample demonstrates how to use the `nRF Cloud REST API`_ to perform Firmware Over-the-Air (FOTA) updates on your device.

Requirements
************

The sample supports the following development kits:

.. table-from-rows:: /includes/sample_board_rows.txt
   :header: heading
   :rows: nrf9160dk_nrf9160_ns

The sample requires an `nRF Cloud`_ account.

.. note::
   This sample requires modem firmware v1.3.x or later.

Overview
********

You can update your device firmware on the `nRF Cloud`_ portal or directly through the `nRF Cloud REST API`_.
See the `nRF Cloud Getting Started FOTA documentation`_ for details.

Limitations
***********

> This sample requires the network carrier to provide date/time to the modem. Without valid date/time the modem can not generate JWTs with an expiration time.

Reviewer: ```suggestion
This sample requires the network carrier to provide date and time to the modem.
Without a valid date and time, the modem cannot generate JWTs with an expiration time.
```

.. _nrf_cloud_rest_fota:

nRF9160: nRF Cloud REST FOTA
#############################

.. contents::
   :local:
   :depth: 2

> The REST FOTA sample demonstrates how to use the `nRF Cloud REST API`_ to perform FOTA updates on your device.

Reviewer: ```suggestion
The REST FOTA sample demonstrates how to use the `nRF Cloud REST API`_ to perform Firmware Over-the-Air (FOTA) updates on your device.
```

If :kconfig:`CONFIG_REST_FOTA_DO_JITP` is enabled, pressing the button configured by :kconfig:`CONFIG_REST_FOTA_BUTTON_EVT_NUM`, when prompted at startup, will request just-in-time provisioning via REST with nRF Cloud.
This is useful when initially provisioning and associating a device on nRF Cloud.
You only need to do this once per device.

On startup, if JITP has not been requested the sample will ask if FOTA should be enabled in the device's shadow. Press the button configured by :kconfig:`CONFIG_REST_FOTA_BUTTON_EVT_NUM` to perform the shadow update.
Depending on how you provisioned your device with nRF Cloud, this step might not be needed. You only need to do this once per device.

If :kconfig:`CONFIG_REST_FOTA_ENABLE_LED` is enabled, an LED configured by :kconfig:`CONFIG_REST_FOTA_LED_NUM` indicates the state of the connection to the LTE network.

Configuration
*************
|config|

Configuration options
=====================

Check and configure the following configuration options for the sample:

> .. option:: CONFIG_REST_FOTA_JOB_CHECK_RATE_MIN - Update check rate
> 
> This configuration option defines how often the sample will check for FOTA updates.

Reviewer: ```suggestion
.. kconfig:: CONFIG_REST_FOTA_JOB_CHECK_RATE_MIN

CONFIG_REST_FOTA_JOB_CHECK_RATE_MIN - Update check rate
   This configuration option defines how often the sample checks for FOTA updates.
```
Format all sample-specific Kconfig options like this. This creates a link from the option mentioned earlier in this doc to this description.

.. option:: CONFIG_REST_FOTA_DL_TIMEOUT_MIN - Download timeout value

This configuration option defines how long to wait for a download to complete.

.. option:: CONFIG_REST_FOTA_ENABLE_LED - Enable LED

This configuration option defines if the LED will be used to indicate connection to the LTE network.

.. option:: CONFIG_REST_FOTA_LED_NUM - Enable LED

> This configuration option defines if the LED to be used, if enabled.

Reviewer: ```suggestion
This configuration option defines if the LED is to be used.
```
